### ISSUE TITLE
Keras dense layer: add support for input with rank greater than 2.

### ISSUE THREAD

**Issue by yanndupis** [tf-encrypted/tf-encrypted#485]
Currently, in TF Encrypted, the Dense layer only supports input with a rank equal to 2. 

However in Keras, the [Dense layer](https://www.tensorflow.org/api_docs/python/tf/keras/layers/Dense) can support input with a rank greater than 2. 

Based on the doc, the input tensor first get flatten. From [here](https://www.tensorflow.org/api_docs/python/tf/keras/layers/Dense):

> Note: if the input to the layer has a rank greater than 2, then it is flattened prior to the initial dot product with kernel.

The output after the matmul [get reshaped](https://github.com/tensorflow/tensorflow/blob/r1.13/tensorflow/python/keras/layers/core.py#L969) back to the original ndim of the input.

So in other words, if input shape is [4, 2, 5] and the dense layer's unit is equal to 4, then the output shape should be [4, 2, 4]. Basically they do a gemm.
